src/components/TextField/TextField.scss

```diff
@@ -0,0 +1,164 @@
+@import "../variables";
+
+$textFieldLabelColor: $charcoal !default;
+
+.uir-textfield {
+    position: relative;
+    display: inline-block;
+    font-family: $font-family;
+    font-size: 13px;
+    font-weight: 400;
+    color: $textFieldLabelColor;
+
+    &:after {
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        display: block;
+        width: 40%;
+        height: 1px;
+        content: "";
+        background: linear-gradient(to right, $grey 0%, transparent 100%);
+        transition: width .25s;
+    }
+
+    &--focus:after {
+        width: 100%;
+    }
+
+    &--invalid:after {
+        background: linear-gradient(to right, $pink 0%, transparent 100%);
+    }
+
+    &--full-width {
+        display: block;
+        width: 100%;
+    }
+
+    &--focus,
+    &--has-value {
+
+        .uir-textfield-label {
+            cursor: default;
+            transform: translate(0, 0);
+        }
+    }
+
+    &--has-right-icon {
+        padding-right: 20px;
+    }
+
+    &--has-left-icon {
+        padding-left: 20px;
+    }
+
+    &--disabled:after,
+    &--readonly:after {
+        background: none;
+    }
+
+    &.uir-textfield--title {
+        font-size: 36px;
+
+        .uir-icon {
+            width: 16px;
+            height: 16px;
+        }
+    }
+
+    &-label-wrapper {
+        position: relative;
+        z-index: 10;
+        display: block;
+        width: 100%;
+        height: 19px;
+    }
+
+    &-label {
+        display: inline-block;
+        font-family: inherit;
+        font-size: 11px;
+        line-height: 1em;
+        text-transform: uppercase;
+        white-space: nowrap;
+        cursor: text;
+        transition: transform .25s;
+        transform: translate(0, 20px);
+
+        &--visually-hidden {
+            position: absolute !important;
+            width: 1px;
+            height: 1px;
+            overflow: hidden;
+            clip: rect(1px, 1px, 1px, 1px);
+        }
+    }
+
+    &-input {
+        width: 100%;
+        padding: 2px 0 3px;
+        font-family: inherit;
+        font-size: inherit;
+        border: 0;
```

Comment:
You removed `outline: none` from here in this commit: 5cce8464d43bb157157e3d130eff698b3234b13c

Are you sure this is correct? Now there's a massive blue highlight on focus and we have the border underline extending on focus so I don't think we need the outline.

Not sure if `outline: none` should be added here or to `:focus`

Reply:
Well spotted, thank you.  Fixed and refined in both TextField and TextArea